lib/Doctrine/ODM/MongoDB/UnitOfWork.php

```diff
@@ -919,7 +919,8 @@ private function computeAssociationChanges($parentDocument, $mapping, $value)
                     $this->scheduleCollectionUpdate($value);
                 }
             }
-            $this->visitedCollections[] = $value;
+            // Inline call to UnitOfWork::markCollectionVisited()
+            $this->visitedCollections[spl_object_hash($value)] = $value;
```

Comment:
I was wondering if we should change `scheduleCollectionFor*`, `isCollectionScheduledFor*` and `unscheduleCollection*` to use `spl_object_hash` as well to get O(log n) instead of O(n) for is/unschedule calls (and stop checking if collection is scheduled before scheduling it) but I was not sure if I won't get myself into hash collisions as it might be happening with EmbeddedDocuments?